components/filter/filter.js

```diff
@@ -276,6 +289,40 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 		`;
 	}
 
+	_clearDimension(dimension) {
+		switch (dimension.type) {
+			case 'd2l-filter-dimension-set': {
+				this.shadowRoot.querySelector(`#${SET_DIMENSION_ID_PREFIX}${dimension.key}`).setSelectionForAll(false);
+				break;
+			}
+		}
+	}
+
+	_clearDimensionData(dimension, sendClearEvent, sendIndividualEvents) {
+		this._totalAppliedCount = this._totalAppliedCount - dimension.appliedCount;
+		dimension.appliedCount = 0;
+
+		switch (dimension.type) {
+			case 'd2l-filter-dimension-set': {
+				dimension.values.forEach(value => {
+					if (value.selected && sendIndividualEvents) {
+						this._dispatchChangeEvent(`${dimension.key}-${value.key}`, { dimension: dimension.key, value: { key: value.key, selected: false } });
+					}
+					value.selected = false;
```

Comment:
For the "manual" clear type, we can't just send the event and clear our counts because when the consumer clears _their_ values, we'll decrease our counts again.  If instead we don't decrease our counts, we'll still be off for all hidden values until the search is cleared.  So we need to decrease the counts and clear all the values we have, and then when the consumer clears them they won't be counted again (because they're trying to set the value to false, which it already is).